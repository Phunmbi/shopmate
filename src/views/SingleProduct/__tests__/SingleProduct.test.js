import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import SingleProductContainer,{ SingleProduct } from '../index';

describe('<ConnectedHomePage />', () => {
  const props = {
    signUp: jest.fn(),
    signIn: jest.fn(),
    searchAllProducts: jest.fn(),
    getCartId: jest.fn(),
    addToCart: jest.fn(),
    retrieveCart: jest.fn(),
    addSingleReview: jest.fn(),
    removeFromCart: jest.fn(),
    updateCart: jest.fn(),
    singleProductDetails: jest.fn(),
    singleProductReviews: jest.fn(),
    match: {
      params: {
        product_id: 1
      }
    },
    cart: [
      {name: 'adeniyi', attributes: "Blue, S", item_id: 1, quantity: 3},
      {name: 'victor', attributes: "Purple, S", item_id: 2, quantity: 4}],
    history: {
      push: jest.fn()
    },
    productDetails: {
      product_id: 1,
      name: 'nigerian',
      subtotal: 455,
      thumbnail:'thumbnail.png',
      image_2: 'image2.png',
      image: 'image.png'
    },
    reviews: [{name: 'named', review: 'Terrible product', rating: 4}],
    cartLoading: false,
    handleSearch: jest.fn,
    allProducts: []
  };

  const initialState = {
    products: {
      products: [
        {name: 'altica', product_id: 12, discounted_price: "0.00"},
        {name: 'italia', product_id: 19},
        {name: 'french', product_id: 17},
        {name: 'nigerian', product_id: 15},
        {name: 'christmas cap', product_id: 14}
      ],
      singleProductDetails: {
        product_id: 1,
        name: 'nigerian',
        subtotal: 455,
        thumbnail:'thumbnail.png',
        image_2: 'image2.png',
        image: 'image.png'
      },
      loading: false,
      reviews: [{name: 'named', review: 'Terrible product', rating: 4}],
      reviewsLoading: false,
      singleProductLoading: false,
    },
    auth: {
      loading: false
    },
    shoppingCart: {
      cart: [{name: 'adeniyi', attributes: "Blue, S", item_id: 1, quantity: 3},
        {name: 'victor', attributes: "Purple, S", item_id: 2, quantity: 4}],
      cart_Id: 4,
      loadingShoppingCart: false
    },
  };
  const mockStore = configureStore();
  const store = mockStore(initialState);
  let wrapper;

  beforeEach(() => {
    wrapper = mount(<SingleProduct {...props} store={store}/>);
    localStorage.clear();
  });

  it('renders appropriately', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should properly render Product with store connections', () => {
    mount(
      <Provider store={store}>
        <MemoryRouter>
          <SingleProductContainer {...props} />
        </MemoryRouter>
      </Provider>
    );
  });

  it('should successfully load both product and review loaders', () => {
    wrapper.setProps({reviewsLoading: true, singleProductLoading: true});

    expect(wrapper.find('Loading').exists()).toEqual(true);
  });

  it('should successfully render signup modal and close it', () => {
    wrapper.setState({displayModal: true, openModal: 'signup'});

    wrapper.find('SignUp').find('.modal-overlay').simulate('click');

    expect(wrapper.find('SignUp').exists()).toEqual(true);
  });

  it('should successfully render signup modal and switch to signin modal', () => {
    wrapper.setState({displayModal: true, openModal: 'signup'});

    wrapper.find('SignUp').find('.additional-options__alternate').find('p').simulate('click');

    expect(wrapper.find('SignIn').exists()).toEqual(true);
  });

  it('should successfully render signup modal and simulate signup', () => {
    wrapper.setState({displayModal: true, openModal: 'signup'});

    wrapper.find('SignUp').find('form').simulate('submit');

    expect(wrapper.find('SignUp').exists()).toEqual(true);
  });

  it('should successfully render signin modal', () => {
    wrapper.setState({displayModal: true, openModal: 'signin'});

    expect(wrapper.find('SignIn').exists()).toEqual(true);
  });

  it('should successfully render signin modal and simulate signin', () => {
    wrapper.setState({displayModal: true, openModal: 'signin'});

    wrapper.find('SignIn').find('form').simulate('submit');

    expect(wrapper.find('SignIn').exists()).toEqual(true);
  });

  it('should successfully render checkout modal', () => {
    wrapper.setState({displayModal: true, openModal: 'checkoutCart'});

    expect(wrapper.find('CheckoutCart').exists()).toEqual(true);
  });

  it('should render ProductDetails and select Colour and size', () => {
    const inst = wrapper.instance();

    wrapper.find('ProductDetails').find('.product-details__sizeButton').simulate('click');

    inst.colourSelector = jest.fn();
    inst.sizeSelector = jest.fn();

    wrapper.find('ProductDetails').find('.product-details__eachColor').simulate('click');
    wrapper.find('ProductDetails').find('.product-details__sizeButton').simulate('click');
    expect(inst.colourSelector).toHaveBeenCalledTimes(0);
    expect(inst.sizeSelector).toHaveBeenCalledTimes(1);
  });

  it('should simulate the quantityClicker ', () => {
    const inst = wrapper.instance();

    wrapper.find('ProductDetails').find('.product-details__quantityClicker').find('img').at(1).simulate('click');
    wrapper.find('ProductDetails').find('.product-details__quantityClicker').find('img').at(0).simulate('click');

    inst.reduceQuantity = jest.fn();
    inst.increaseQuantity = jest.fn();

    wrapper.find('ProductDetails').find('.product-details__quantityClicker').find('img').at(1).simulate('click');
    wrapper.find('ProductDetails').find('.product-details__quantityClicker').find('img').at(0).simulate('click');

    expect(inst.increaseQuantity).toHaveBeenCalledTimes(0);
    expect(inst.reduceQuantity).toHaveBeenCalledTimes(1);
  });

  it('should simulate reduce quantity when quantity is 1', function () {
    const inst = wrapper.instance();

    wrapper.find('ProductDetails').find('.product-details__quantityClicker').find('img').at(0).simulate('click');

    inst.reduceQuantity = jest.fn();
    wrapper.find('ProductDetails').find('.product-details__quantityClicker').find('img').at(0).simulate('click');

    expect(inst.reduceQuantity).toHaveBeenCalledTimes(0);
  });

  it('should simulate add to cart', () => {
    const inst = wrapper.instance();
    wrapper.find('ProductDetails').find('.product-details__action').find('button').simulate('click');

    localStorage.setItem("cart_id", 'idCart');
    inst.handleAddToCart = jest.fn();

    wrapper.find('ProductDetails').find('.product-details__action').find('button').simulate('click');

    expect(wrapper.find('ProductDetails').exists()).toEqual(true);
  });

  it('should simulate updateCart', () => {
    //Set in test local storage mocks
    localStorage.setItem('1', JSON.stringify({quantity: 3}));
    localStorage.setItem('2', JSON.stringify({quantity: 4}));
    localStorage.setItem('cart_id', '3');

    wrapper.setState({displayModal: true, openModal: 'checkoutCart'});
    wrapper.find('CheckoutCart').setState({ changingQuantity: true , 1: {quantity: 3}, 2: {quantity: 4}});
    const inst = wrapper.instance();
    wrapper.find('.checkout-container__quantity').find('img').at(0).simulate('click');
    wrapper.find('.checkout-container__quantity').find('img').at(0).simulate('click');
    wrapper.find('.checkout-container__quantity').find('img').at(0).simulate('click');
    wrapper.find('.checkout-container__quantity').find('img').at(1).simulate('click');

    wrapper.find('.active-button').simulate('click');

    inst.handleUpdateCart = jest.fn();
    wrapper.find('.active-button').simulate('click');

    expect(wrapper.find('CheckoutCart').exists()).toEqual(true);
  });

  it('should simulate updateCart and simulate checkout, while logged in', () => {
    //Set in test local storage mocks
    localStorage.setItem('1', JSON.stringify({quantity: 3}));
    localStorage.setItem('2', JSON.stringify({quantity: 4}));
    localStorage.setItem('isAuthenticated', 'true');
    localStorage.setItem('cart_id', '3');

    const inst = wrapper.instance();

    wrapper.setState({displayModal: true, openModal: 'checkoutCart'});
    wrapper.find('CheckoutCart').setState({ changingQuantity: true , 1: {quantity: 3}, 2: {quantity: 4}});
    wrapper.find('.checkout-container__quantity').find('img').at(0).simulate('click');
    wrapper.find('.checkout-container__quantity').find('img').at(0).simulate('click');
    wrapper.find('.checkout-container__quantity').find('img').at(0).simulate('click');
    wrapper.find('.checkout-container__quantity').find('img').at(1).simulate('click');

    wrapper.find('.checkout-button').find('button').at(2).simulate('click');

    inst.startCheckout = jest.fn();
    wrapper.find('.checkout-button').find('button').at(2).simulate('click');

    expect(wrapper.find('CheckoutCart').exists()).toEqual(true);
  });

  it('should simulate updateCart and simulate checkout while logged out', () => {
    //Set in test local storage mocks
    localStorage.setItem('1', JSON.stringify({quantity: 3}));
    localStorage.setItem('2', JSON.stringify({quantity: 4}));
    localStorage.setItem('cart_id', '3');

    const inst = wrapper.instance();

    wrapper.setState({displayModal: true, openModal: 'checkoutCart'});
    wrapper.find('CheckoutCart').setState({ changingQuantity: true , 1: {quantity: 3}, 2: {quantity: 4}});
    wrapper.find('.checkout-container__quantity').find('img').at(0).simulate('click');
    wrapper.find('.checkout-container__quantity').find('img').at(0).simulate('click');
    wrapper.find('.checkout-container__quantity').find('img').at(0).simulate('click');
    wrapper.find('.checkout-container__quantity').find('img').at(1).simulate('click');

    wrapper.find('.checkout-button').find('button').at(2).simulate('click');

    inst.startCheckout = jest.fn();

    expect(wrapper.find('CheckoutCart').exists()).toEqual(false);
  });

  it('should simulate handleAddReviews', () => {
    const inst = wrapper.instance();

    wrapper.find('ProductReviews').find('AddReview').find('.add-review__submit').find('input').simulate('click');

    localStorage.setItem("isAuthenticated", "true");
    inst.handleAddSingleReview = jest.fn();
    wrapper.find('ProductReviews').find('AddReview').find('.add-review__submit').find('input').simulate('click');

    expect(wrapper.find('AddReview').exists()).toEqual(true);
  });

  it('should simulate the handleChange in Add Reviews', () => {
    wrapper.find('ProductReviews').find('AddReview').find('.add-review__reviewDetails').find('textarea').simulate('change');

    expect(wrapper.find('AddReview').exists()).toEqual(true);
  });

  it('should simulate the handleRating in Add Reviews silverStars', () => {
    wrapper.find('ProductReviews').find('AddReview').find('.add-review__stars').find('img').at(1).simulate('click');

    expect(wrapper.find('AddReview').exists()).toEqual(true);
  });

  it('should simulate the handleRating in Add Reviews Gold stars', () => {
    const inst = wrapper.find('AddReview').instance();
    inst.setState({
      values: {
        nickname: "",
        review: "",
        rating: 0,
        stars: {
          1: "gold",
          2: "gold",
          3: "silver",
          4: "silver",
          5: "silver",
        }
      }
    });
    wrapper.find('ProductReviews').find('AddReview').find('.add-review__stars').find('img').at(1).simulate('click');

    expect(wrapper.find('AddReview').exists()).toEqual(true);
  });


  it('should simulate the else on displaying reviews component in ProductReviews', () => {
    wrapper.setProps({reviews: []});

    expect(wrapper.find('ProductReviews').exists()).toEqual(true);
  });

  it('should simulate singleProductLoading component in ProductDetails', () => {
    wrapper.setProps({
      productDetails: {
        product_id: 1,
        name: 'nigerian',
        subtotal: 455,
        thumbnail:'thumbnail.png',
        image_2: 'image2.png',
        image: 'image.png',
        discounted_price: "0.00"
      }
    });
    wrapper.find('ProductDetails').find('.product-details__thumbnails').find('img').at(0).simulate('mouseOver');
    wrapper.find('ProductDetails').find('.product-details__thumbnails').find('img').at(1).simulate('mouseOver');

    expect(wrapper.find('ProductDetails').exists()).toEqual(true);
  });

  it('should simulate through NavBarCollapse redirection to homePage', () => {
    localStorage.setItem("isAuthenticated", "true");

    wrapper.find('.navbar-collapse__container').find('h3').simulate('click');

    expect(wrapper.exists()).toEqual(true);
  });

  it('should simulate through NavBarCollapse and simulate renderDropDown menu', () => {
    localStorage.setItem("isAuthenticated", "true");
    wrapper.find('.navbar-collapse__container').find('img').simulate('click');

    wrapper.find('.dropdown-main').find('p').at(0).simulate('click');

    expect(wrapper.find('NavBarCollapse').exists()).toEqual(true);
  });

  it('should simulate through NavBarCollapse redirection to profile page', () => {
    localStorage.setItem("isAuthenticated", "true");

    wrapper.find('.navbar-collapse__container').find('img').simulate('click');
    wrapper.find('.navbar-collapse__container').find('img').simulate('click');
    wrapper.find('.navbar-collapse__container').find('img').simulate('click');

    wrapper.find('.dropdown-main').find('p').at(1).simulate('click');

    expect(wrapper.find('NavBarCollapse').exists()).toEqual(true);
  });
});