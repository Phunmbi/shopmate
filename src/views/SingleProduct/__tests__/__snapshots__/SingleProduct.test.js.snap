// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConnectedHomePage /> renders appropriately 1`] = `
<SingleProduct
  addSingleReview={[MockFunction]}
  addToCart={[MockFunction]}
  allProducts={Array []}
  cart={
    Array [
      Object {
        "attributes": "Blue, S",
        "item_id": 1,
        "name": "adeniyi",
        "quantity": 3,
      },
      Object {
        "attributes": "Purple, S",
        "item_id": 2,
        "name": "victor",
        "quantity": 4,
      },
    ]
  }
  cartLoading={false}
  getCartId={[MockFunction]}
  handleSearch={[Function]}
  history={
    Object {
      "push": [MockFunction],
    }
  }
  match={
    Object {
      "params": Object {
        "product_id": 1,
      },
    }
  }
  productDetails={
    Object {
      "image": "image.png",
      "image_2": "image2.png",
      "name": "nigerian",
      "product_id": 1,
      "subtotal": 455,
      "thumbnail": "thumbnail.png",
    }
  }
  removeFromCart={[MockFunction]}
  retrieveCart={
    [MockFunction] {
      "calls": Array [
        Array [
          null,
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
  reviews={
    Array [
      Object {
        "name": "named",
        "rating": 4,
        "review": "Terrible product",
      },
    ]
  }
  searchAllProducts={[MockFunction]}
  signIn={[MockFunction]}
  signUp={[MockFunction]}
  singleProductDetails={
    [MockFunction] {
      "calls": Array [
        Array [
          undefined,
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
  singleProductReviews={
    [MockFunction] {
      "calls": Array [
        Array [
          undefined,
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  updateCart={[MockFunction]}
>
  <div
    className="product"
  >
    <header>
      <NavbarHome
        bagTotal={[Function]}
        cartCount={2}
        handleDisplayModal={[Function]}
        retrieveCart={[Function]}
      >
        <div
          className="navbar-home"
        >
          <div
            className="navbar-home__main"
          >
            <div
              className="navbar-home__action .Text-Style-2"
            >
              <h3
                className="navbar-home__action1"
              >
                Hi! 
              </h3>
              <h3
                className="navbar-home__action2"
                onClick={[Function]}
              >
                Sign in
              </h3>
              <h3
                className="navbar-home__action1"
              >
                or 
              </h3>
              <h3
                className="navbar-home__action2"
                onClick={[Function]}
              >
                Register
              </h3>
            </div>
            <div
              className="navbar-home__cart"
              onClick={[Function]}
            >
              <Cart
                cartCount={2}
              >
                <div
                  className="cart-main"
                >
                  <img
                    alt="Checkout Bag"
                    src="cart-white.svg"
                  />
                  <div
                    className="cart-main__count"
                  >
                    2
                  </div>
                </div>
              </Cart>
              <div>
                Your Bag: £
                NaN
              </div>
            </div>
          </div>
        </div>
      </NavbarHome>
      <NavBarCollapse
        history={
          Object {
            "push": [MockFunction],
          }
        }
      >
        <div
          className="navbar-collapse"
        >
          <div
            className="navbar-collapse__container"
          >
            <h3
              onClick={[Function]}
            >
              SHOPMATE
            </h3>
          </div>
        </div>
      </NavBarCollapse>
    </header>
    <section
      className="product-details"
    >
      <ProductDetails
        averageRating={4}
        colourSelector={[Function]}
        handleAddToCart={[Function]}
        handleDisplayModal={[Function]}
        increaseQuantity={[Function]}
        productDetails={
          Object {
            "image": "image.png",
            "image_2": "image2.png",
            "name": "nigerian",
            "product_id": 1,
            "subtotal": 455,
            "thumbnail": "thumbnail.png",
          }
        }
        quantity={1}
        reduceQuantity={[Function]}
        selectedColour="teal"
        selectedSize="S"
        sizeSelector={[Function]}
      >
        <div
          className="product-details__main"
        >
          <div
            className="product-details__container"
          >
            <div
              className="product-details__imageContainer"
            >
              <div
                className="product-details__mainImage"
              >
                <img
                  alt=""
                  src="https://backendapi.turing.com/images/products/null"
                />
              </div>
              <div
                className="product-details__thumbnails"
              >
                <img
                  alt=""
                  onMouseOver={[Function]}
                  src="https://backendapi.turing.com/images/products/thumbnail.png"
                />
                <img
                  alt=""
                  onMouseOver={[Function]}
                  src="https://backendapi.turing.com/images/products/image2.png"
                />
              </div>
            </div>
            <div
              className="product-details__details"
            >
              <div
                className="product-details__ratings"
              >
                <img
                  alt=""
                  key="0"
                  src="gold 4.svg"
                />
                <img
                  alt=""
                  key="1"
                  src="gold 4.svg"
                />
                <img
                  alt=""
                  key="2"
                  src="gold 4.svg"
                />
                <img
                  alt=""
                  key="3"
                  src="gold 4.svg"
                />
                <img
                  alt=""
                  key="4"
                  src="grey 1.svg"
                />
              </div>
              <h3>
                nigerian
              </h3>
              <div
                className="product-details__prices"
              >
                <h3
                  className="product-details__discountedPrice"
                >
                  £
                </h3>
                <h3
                  className="product-details__price"
                >
                  £
                </h3>
              </div>
              <div
                className="product-details__color"
              >
                <h4>
                  Color
                </h4>
                <div
                  className="product-details__eachColor"
                  onClick={[Function]}
                >
                  <img
                    alt="blue"
                    key="blue"
                    src="blue.svg"
                  />
                  <img
                    alt="teal"
                    className="product-details__selectedColor"
                    key="teal"
                    src="teal.svg"
                  />
                  <img
                    alt="red"
                    key="red"
                    src="red.svg"
                  />
                  <img
                    alt="orange"
                    key="orange"
                    src="orange.svg"
                  />
                  <img
                    alt="yellow"
                    key="yellow"
                    src="yellow.svg"
                  />
                  <img
                    alt="green"
                    key="green"
                    src="green.svg"
                  />
                  <img
                    alt="purple"
                    key="purple"
                    src="purple.svg"
                  />
                </div>
              </div>
              <div
                className="product-details__sizeMenu"
              >
                <div
                  className="product-details__sizes"
                >
                  <h3>
                    Size
                  </h3>
                  <p>
                    Size guide
                  </p>
                </div>
                <div
                  className="product-details__sizeButton"
                  onClick={[Function]}
                >
                  <div
                    className="product-details__eachSize"
                    key="XS"
                  >
                    XS
                  </div>
                  <div
                    className="product-details__eachSizeSelected"
                    key="S"
                  >
                    S
                  </div>
                  <div
                    className="product-details__eachSize"
                    key="M"
                  >
                    M
                  </div>
                  <div
                    className="product-details__eachSize"
                    key="L"
                  >
                    L
                  </div>
                  <div
                    className="product-details__eachSize"
                    key="XL"
                  >
                    XL
                  </div>
                  <div
                    className="product-details__eachSize"
                    key="XXL"
                  >
                    XXL
                  </div>
                </div>
              </div>
              <div
                className="product-details__quantity"
              >
                <h3>
                  Quantity
                </h3>
                <div
                  className="product-details__quantityClicker"
                >
                  <img
                    alt=""
                    onClick={[Function]}
                    src="minus.svg"
                  />
                  <div
                    className="product-details__count"
                  >
                    <p>
                      1
                    </p>
                  </div>
                  <img
                    alt=""
                    onClick={[Function]}
                    src="plus.svg"
                  />
                </div>
              </div>
              <div
                className="product-details__action"
              >
                <button
                  onClick={[Function]}
                >
                  Add to Cart
                </button>
                <div
                  className="product-details__wishList"
                >
                  <img
                    alt=""
                    src="heart.svg"
                  />
                  <p>
                    Add to Wish List
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </ProductDetails>
    </section>
    <section
      className="product-reviews"
    >
      <ProductReviews
        handleAddSingleReview={[Function]}
        productDetails={
          Object {
            "image": "image.png",
            "image_2": "image2.png",
            "name": "nigerian",
            "product_id": 1,
            "subtotal": 455,
            "thumbnail": "thumbnail.png",
          }
        }
        reviews={
          Array [
            Object {
              "name": "named",
              "rating": 4,
              "review": "Terrible product",
            },
          ]
        }
      >
        <div
          className="product-reviews__main"
        >
          <div
            className="product-reviews__container"
          >
            <div
              className="product-reviews__width"
            >
              <div
                className="product-reviews__reviewsSection"
              >
                <div
                  className="product-reviews__title"
                >
                  <h3>
                    Product reviews
                  </h3>
                </div>
                <SingleProductReview
                  key="0"
                  review={
                    Object {
                      "name": "named",
                      "rating": 4,
                      "review": "Terrible product",
                    }
                  }
                >
                  <div
                    className="single-review__container"
                  >
                    <div
                      className="single-review__rating"
                    >
                      <img
                        alt=""
                        key="0"
                        src="gold 4.svg"
                      />
                      <img
                        alt=""
                        key="1"
                        src="gold 4.svg"
                      />
                      <img
                        alt=""
                        key="2"
                        src="gold 4.svg"
                      />
                      <img
                        alt=""
                        key="3"
                        src="gold 4.svg"
                      />
                      <img
                        alt=""
                        key="4"
                        src="grey 1.svg"
                      />
                    </div>
                    <div
                      className="single-review__description"
                    >
                      <p>
                        Terrible product
                      </p>
                    </div>
                  </div>
                  <div
                    className="single-review__details"
                  >
                    <div
                      className="single-review__reviewer"
                    >
                      <h3>
                        named
                      </h3>
                    </div>
                    <div
                      className="single-review__reviewedOn"
                    >
                      <p>
                        a few seconds ago
                      </p>
                    </div>
                  </div>
                </SingleProductReview>
              </div>
              <div
                className="product-reviews__add"
              >
                <AddReview
                  handleAddSingleReview={[Function]}
                  productDetails={
                    Object {
                      "image": "image.png",
                      "image_2": "image2.png",
                      "name": "nigerian",
                      "product_id": 1,
                      "subtotal": 455,
                      "thumbnail": "thumbnail.png",
                    }
                  }
                >
                  <div
                    className="add-review__container"
                  >
                    <h3>
                      Add a review
                    </h3>
                    <form>
                      <div
                        className="add-review__reviewDetails"
                      >
                        <label
                          htmlFor="review"
                        >
                          Your review
                          <div>
                            <textarea
                              cols="30"
                              id="review"
                              name="review"
                              onChange={[Function]}
                              rows="10"
                            />
                            <p>
                              Your review must be at least 50 characters 
                              <span>
                                Full review guidelines
                              </span>
                            </p>
                          </div>
                        </label>
                      </div>
                      <div
                        className="add-review__rating"
                      >
                        <label
                          htmlFor="rating"
                        >
                          Overall rating
                          <div
                            className="add-review__stars"
                          >
                            <img
                              alt=""
                              key="0"
                              name={0}
                              onClick={[Function]}
                              src="grey 1.svg"
                            />
                            <img
                              alt=""
                              key="1"
                              name={1}
                              onClick={[Function]}
                              src="grey 1.svg"
                            />
                            <img
                              alt=""
                              key="2"
                              name={2}
                              onClick={[Function]}
                              src="grey 1.svg"
                            />
                            <img
                              alt=""
                              key="3"
                              name={3}
                              onClick={[Function]}
                              src="grey 1.svg"
                            />
                            <img
                              alt=""
                              key="4"
                              name={4}
                              onClick={[Function]}
                              src="grey 1.svg"
                            />
                          </div>
                        </label>
                      </div>
                      <div
                        className="add-review__submit"
                      >
                        <input
                          onClick={[Function]}
                          type="submit"
                          value="Submit"
                        />
                      </div>
                    </form>
                  </div>
                </AddReview>
              </div>
            </div>
          </div>
        </div>
      </ProductReviews>
    </section>
    <footer
      className="footer"
    >
      <Footer>
        <div
          className="footer-main"
        >
          <div
            className="footer-container"
          >
            <div
              className="footer-container__categories"
            >
              <h3>
                Women
              </h3>
              <h3>
                Men
              </h3>
              <h3>
                Kids
              </h3>
              <h3>
                Shoes
              </h3>
              <h3>
                Brands
              </h3>
            </div>
            <div
              className="footer-container__social"
            >
              <div
                className="footer-container__socialIcons"
              >
                <img
                  alt="instagram"
                  src="instagramIcon.svg"
                />
              </div>
              <div
                className="footer-container__socialIcons"
              >
                <img
                  alt="pinterest"
                  src="pinterestIcon.svg"
                />
              </div>
              <div
                className="footer-container__socialIcons"
              >
                <img
                  alt="twitter"
                  src="twitterIcon.svg"
                />
              </div>
              <div
                className="footer-container__socialIcons"
              >
                <img
                  alt="facebook"
                  src="facebookIcon.svg"
                />
              </div>
            </div>
            <div
              className="footer-container__information"
            >
              © shopmate Ltd • Contact • Privacy Policy
            </div>
          </div>
        </div>
      </Footer>
    </footer>
  </div>
</SingleProduct>
`;
